enumerables.rb

```diff
@@ -0,0 +1,143 @@
+# Defining Enumerable
+module Enumerable
+  def my_each
+    pos = 0
+    return to_enum unless block_given?
+
+    loop do
+      break if pos >= length
+
+      yield self[pos]
+      pos += 1
+    end
+    self
+  end
+
+  # Defining my_each_with_index
+  def my_each_with_index
+    pos = 0
+    return to_enum unless block_given?
+
+    loop do
+      break if pos >= length
+
+      yield self[pos], pos
+      pos += 1
+    end
+    self
+  end
+
+  # Defining my_select
+  def my_select
+    return_arr = []
+    return to_enum unless block_given?
+
+    my_each do |element|
+      return_arr.push(element) if yield element
+    end
+
+    return_arr
+  end
+
+  # Defining my_all
+  def my_all?(arg = nil)
+    return_val = true
+    my_each do |element|
+      return_val = (block_given? && yield(element)) || element.poitive?(arg)
+      return_val = true?(element) if !block_given? && !arg
```

Comment:
I think you should read this [article](https://medium.com/@josh.pann/conditional-assignment-operator-in-ruby-ac0ebc740ced) to understand how the conditional assignment operator works (on line 46).

But still, you are assigning the same variable twice (l.46 & 47) and one is overwriting the other. That is causing a problem with your method when you pass a block, a regular element (let's say a number), or a Class: `wrong number of arguments (given 1, expected 0)`

Reply:
Since i have added === which need to catch the following & is returning the arguments, kindly recheck.